### ISSUE TITLE
Cannot style dropdown using `overlayStyle` or `overlayClassName`

### ISSUE THREAD

**Issue by IonelLupu** [ant-design/ant-design#39324]
### Reproduction link

[https://stackblitz.com/edit/react-g24p3b?file=demo.tsx](https://stackblitz.com/edit/react-g24p3b?file=demo.tsx)
![image](https://user-images.githubusercontent.com/4083652/205978120-20bf456c-f4b5-48e7-99bc-09486e655027.png)

### Steps to reproduce

- add an `overlayStyle` (or `overlayClassName`) to a dropdown with `{marginTop: '50px'}`. 
- open de dropdown
- observe the dropdown not being "50px" down

### What is expected?

For the dropdown to be affected by the styles/classes

### What is actually happening?

Ant design seems it's calculating the position of the dropdown including the added styles. I think the styles should be added AFTER the rendering of the dropdown. I am saying this because adding the styles (`marginTop`) manually in the Devtools window affects the dropdown as expected.

| Environment | Info |
| --- | --- |
| antd | 5.0.4 |
| React | 18.2.0 |
| System | MacOS Ventura 13.0.1 |
| Browser | Chrome 108 |

**Comment by afc163** [comment]
See https://github.com/ant-design/ant-design/issues/11942#issuecomment-417195388

**Comment by afc163** [comment]
Please search FAQ issues before create a new issue.

**Comment by IonelLupu** [comment]
hey @afc163 Thank you for the tip. It works ❤️ . 

I was checking the dropdown docs and I didn't see anything related to the `align` property. I also checked the [FAQ](https://ant.design/docs/react/faq) page and I didn't find anything related to it either. I actually found something on the [Tooltip](https://ant.design/components/tooltip#API) page.

Maybe we can add the `align` property in the docs as well for Dropdown and Autocomplete. For Autocomplete I found this  undocumented property `dropdownAlign`
